CmdStarter.Lib/Attributes/AutoComplete/AutoCompleteAttribute{T}.cs

```diff
@@ -0,0 +1,91 @@
+﻿using com.cyberinternauts.csharp.CmdStarter.Lib.Interfaces;
+using System.CommandLine.Completions;
+
+namespace com.cyberinternauts.csharp.CmdStarter.Lib.Attributes
+{
+    /// <inheritdoc cref="AutoCompleteAttribute"/>
+    /// <typeparam name="T">
+    /// Defines the behaviour of the attribute.
+    /// <para>
+    /// More info in constructors.
+    /// </para>
+    /// </typeparam>
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Parameter, AllowMultiple = true)]
+    public sealed class AutoCompleteAttribute<T> : AutoCompleteAttribute
+    {
+        private readonly IAutoCompleteFactory? _factory;
+
+        /// <summary>
+        /// <para>
+        /// If <typeparamref name="T"/> is <see langword="typeof"/> <see cref="Enum"/>
+        /// generates auto completions from all values. 
+        /// </para>
+        /// <para>
+        /// If <typeparamref name="T"/> is <see langword="typeof"/> <see cref="IAutoCompleteProvider"/>
+        /// retrieves auto completions from there.
+        /// </para>
+        /// </summary>
+        public AutoCompleteAttribute()
+            : base(typeof(T))
+        { }
+
+        /// <inheritdoc cref="AutoCompleteAttribute(object[])"/>
+        public AutoCompleteAttribute(params object[] completions)
+            : base(completions)
+        {
+            _factory = GetFactory();
+        }
+
+        /// <inheritdoc cref="AutoCompleteAttribute(Type)"/>
+        public AutoCompleteAttribute(Type type)
+            : base(type)
+        {
+            _factory = GetFactory();
+        }
```

Comment:
Are you sure those constructors should exist? Because I could use `[AutoComplete<T1>(typeof(T1))]` which is... special and confusing?

Reply:
Although what you did is confusing it would throw an error as `T` is neither an enum or an `AutoCompleteProvider` (unless it implements those interfaces too in which case there's nothing wrong with it).

This constructor is there to be able to run a factory on enums or elements from a provider.

Reply:
From the changes of this class to be generic facade over its non-generic one, the `(Type type)` constructor isn't useful anymore, but the other yes. I saw it afterward, but didn't want to change the lines selection of the comment (hahaha, maybe another test for you 👍 ).

Reply:
I merged the two classes' functionality into `AutoCompleteAttribute` so now `AutoCompleteAttribute<T>`is more like a generic wrapper.